### ISSUE TITLE
SafeAreaUtils should include `systemBars()` or `captionBar()` inset type in android

### ISSUE THREAD

**Issue by jorgegil96** [th3rdwave/react-native-safe-area-context#509]
Otherwise the top of the view will be cutoff by caption bars in freeform mode, since AOSP uses `captionBar()` InsetType for captions (title bar). 

Repro steps:
Using phone with Android R+ or Android Studio's AVD Desktop emulator:
1. (skip if using desktop AVD) In android R+, enable Freeform in Settings > System > Developer Options > Enable Freeform Windows
2. Open a view using SafeAreaUtils, such as a SafeAreaProvider
3. (skip if using desktop AVD) Go to Android Overview, tap the App Icon, click on "Freeform" to enter freeform mode
4. The top portion of the app is occluded by the system caption/title bar

I'm using Discord to test, as it uses a SafeAreaProvider.

```
//SafeAreaUtils.kt

@RequiresApi(Build.VERSION_CODES.R)
private fun getRootWindowInsetsCompatR(rootView: View): EdgeInsets? {
  val insets =
      rootView.rootWindowInsets?.getInsets(
          WindowInsets.Type.statusBars() or
              WindowInsets.Type.displayCutout() or
              WindowInsets.Type.navigationBars())
          ?: return null
  return EdgeInsets(
      top = insets.top.toFloat(),
      right = insets.right.toFloat(),
      bottom = insets.bottom.toFloat(),
      left = insets.left.toFloat())
}
```

The above should be changed to:
```
val insets =
      rootView.rootWindowInsets?.getInsets(WindowInsets.Type.systemBars() or WindowInsets.Type.displayCutout())
          ?: return null
```
or
```
val insets =
      rootView.rootWindowInsets?.getInsets(
          WindowInsets.Type.statusBars() or
              WindowInsets.Type.displayCutout() or
              WindowInsets.Type.navigationBars() or
              WindowInsets.Type.captionBar())
          ?: return null
```

See:
https://developer.android.com/reference/android/view/WindowInsets.Type#captionBar() https://developer.android.com/reference/android/view/WindowInsets.Type#systemBars()
https://chromeos.dev/en/posts/desktop-avd-in-android-studio
![Screenshot 2024-07-08 at 1 12 29 p m](https://github.com/th3rdwave/react-native-safe-area-context/assets/9954667/ff786255-6b80-4a8e-bbec-4048ae144bc4)
![Screenshot 2024-07-08 at 1 13 31 p m](https://github.com/th3rdwave/react-native-safe-area-context/assets/9954667/9b39798d-4bb4-4d9c-9448-1a5ff97e8517)

**Comment by janicduplessis** [comment]
Would you like to open a PR to make this change?

**Comment by jorgegil96** [comment]
I'm not really a react dev so I'll need to find time to get set up and all that. Figured it might be easier for someone else to do it if they're already active devs given it's a small tweak.

**Comment by janicduplessis** [comment]
No problem, I will look at it then!

**Comment by janicduplessis** [comment]
Fixed in 4.10.8
